htdocs/compta/facture.php

```diff
@@ -611,7 +611,7 @@
 			$i = 0;
 			foreach ($object->lines as $line)
 			{
-				if ($line->total_ht!=0)
+				if (empty($line->special_code) && $line->total_ht != 0) // Remove lines with special code
```

Comment:
Current known special_code are 
1=frais de port
2=ecotax
In both cases, we need to include the content into the converted discount

Which cas do you have to exclude them ?

Reply:
@eldy 
"special_code" is used by external module, the value of "special_code" is the module number.
and for my module I also use the field "product_type" with the value 9 for this type of lines. Maybe it is necessary to use "product_type" instead?

Reply:
I think a lot of modules has defined a different rule for the field special_code.
I think (in fact i am sure) that some modules does not use a module number here. I found another use cas: 3 is used for options lines by core code itfself.
We can defined a rule like special_code < 100 is used for discount convertion, > 100 not.
But if hava solution to detect your line using product_type, it is better...